powa_collector/powa_worker.py

```diff
@@ -400,10 +400,10 @@ def __worker_main(self):
                     self.logger.error("Error during snapshot: %s" % e)
                     if (self.__repo_conn is None
                             or self.__repo_conn.closed > 0):
-                        self.__repo_conn = None
```

Comment:
mmm, actually the first test is a bit cryptic.  Should we change the test for "is not None and..."?

Then, according to psycopg2 doc, if connection.closed > 0, then it's closed or broken.  I'm assuming that "broken" actually means that there's still a backend opened?

Reply:
I think this test is good. What do you think is a problem ?

Reply:
If the repository connection is None, then by definition there's no cleanup to do (and __disconnect_repo() will check that and do nothing in that case).  Here we just need to check that there's an object, so we don't try to access an unexisting property.